src/derivers/shared.ts

```diff
@@ -223,13 +227,15 @@ async function deriveNode(
  *
  * @param options - The options for deriving a secret extension.
  * @param options.privateKey - The parent private key bytes.
+ * @param options.publicKey - The parent public key bytes.
```

Comment:
Probably worth documenting that this is compressed public key

Reply:
Would've prevented some debugging 😅 That's why I added the check as well.